### PR TITLE
Automatically Convert `name` Argument and `registry`  Argument to Lowercase in ImageSpec

### PR DESCRIPTION
**Pull request by Future-Outlier**
# TL;DR
I recently started using "ImageSpec" and encountered an issue when I provided a name argument that wasn't in lowercase. Currently, the "ImageSpec" requires the `name` argument to be in lowercase, and it doesn't handle cases where users provide uppercase or mixed-case strings.

To enhance user experience and reduce potential errors, I propose that we automatically convert the `name` argument to lowercase within the "ImageSpec". This minor adjustment will make the library more user-friendly and save users' time by preventing them from repeating this common mistake.

Please consider this improvement for a future release. Looking forward to your feedback!

## Type
 - [ ] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

### CODE REVIEW

**Review by pingsutw — COMMENTED** [review]
could you use post_init instead?

```python
class ImageSpec():
    ...
    ...

    def __post_init__(self):
        self.name = self.name.lower()
        self.registry = self.registry.lower()
```

**Comment by Future-Outlier** [comment]
> could you use post_init instead?
> 
> ```python
> class ImageSpec():
>     ...
>     ...
> 
>     def __post_init__(self):
>         self.name = self.name.lower()
>         self.registry = self.registry.lower()
> ```

Amazing advice, thanks a lot!

**Comment by Future-Outlier** [comment]
Due to this error, I will check if `name` and `registry` is `None` or not.
image

**Review by pingsutw — APPROVED** [review]

**Review by pingsutw — APPROVED** [review]
